Code/Framework/AzCore/AzCore/std/string/string_view.h

```diff
@@ -309,78 +309,87 @@ namespace AZStd
-            { 
-                return __builtin_memcmp(s1, s2, count); 
-            } 
-            else if constexpr (AZStd::is_same_v<char_type, wchar_t>) 
-            { 
-                return __builtin_wmemcmp(s1, s2, count); 
-            } else 
-#endif 
-            { 
-                if (az_builtin_is_constant_evaluated()) 
-                { 
-                    for (; count; --count, ++s1, ++s2) 
+        static constexpr int compare(const char_type* s1, const char_type* s2, size_t count) noexcept
+        {
+            // In GCC versions prior to major version 10, __builtin_memcmp fails in valid checks in constexpr evaluation
+#if !defined(AZ_COMPILER_GCC) || AZ_COMPILER_GCC >= 100000
+            if constexpr (AZStd::is_same_v<char_type, char>)
+            {
+                return __builtin_memcmp(s1, s2, count);
+            }
+            else if constexpr (AZStd::is_same_v<char_type, wchar_t>)
+            {
+                return __builtin_wmemcmp(s1, s2, count);
+            }
+            else
+#endif
+            {
+                if (az_builtin_is_constant_evaluated())
+                {
+                    for (; count; --count, ++s1, ++s2)
                     {
-                        if (lt(*s1, *s2)) 
-                        { 
-                            return -1; 
-                        } 
-                        else if (lt(*s2, *s1)) 
-                        { 
-                            return 1; 
-                        } 
-                    } 
-                    return 0; 
-                } 
-                else 
-                { 
-                    return ::memcmp(s1, s2, count * sizeof(char_type)); 
-                } 
-            } 
+                        if (lt(*s1, *s2))
+                        {
+                            return -1;
+                        }
+                        else if (lt(*s2, *s1))
+                        {
+                            return 1;
+                        }
+                    }
+                    return 0;
+                }
+                else
+                {
+                    return ::memcmp(s1, s2, count * sizeof(char_type));
+                }
+            }
         }
 
         static constexpr size_t length(const char_type* s) noexcept
         {
             // For GCC versions less than 10, __builtin_strlen and __builtin_wcslen is not supported as const expressions
             // so for that case it will need to manually count the characters (at compile time) instead
-#if defined(AZ_COMPILER_GCC) && AZ_COMPILER_GCC < 100000
 
             if constexpr (AZStd::is_same_v<char_type, char>)
             {
+#if defined(AZ_COMPILER_GCC) && AZ_COMPILER_GCC < 100000
                 if (!az_builtin_is_constant_evaluated())
                 {
                     return strlen(s);
                 }
+                else
+                {
+                    size_t strLength{};
```

Comment:
Should be  `size_t strLength{ 0 };` at least

Reply:
It is my friend.
[Zero initialization](https://en.cppreference.com/w/cpp/language/zero_initialization)

Reply:
I should've known 0️⃣